### PR TITLE
Change the  kotlin formatter to better support building under Java 17

### PR DESCRIPTION
**Pull request by evanchooly**
This updates the spotless plugin to use the `ktfmt` formatter instead of `ktlint` since the latter doesn't yet seem to work on Java 17.

### CODE REVIEW

**Comment by geoand** [comment]
This is almost impossible to review unfortunately due to the size - obviously for no fault of your own

File: build-parent/pom.xml [hunk 1]
```diff
@@ -840,12 +840,16 @@
 
                         <configuration>
                             <kotlin>
+                                <ktfmt>
+                                    <style>KOTLINLANG</style>
```
Comment (gsmet) [inline build-parent/pom.xml:844]: Dumb question: do the other styles generate code that are more like we used to have?

Comment (evanchooly) [reply build-parent/pom.xml:844]: i played with the others and they're all ending up touching quite a few files even it's only changing indentation.  But mostly it's variance in wrapping rules and stacked annotations vs all one on line kinds of things.

**Comment by evanchooly** [comment]
are we good here, then?

**Comment by geoand** [comment]
Lack of time likely ☹️. I'll try and have a look tomorrow

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
I rebased just to make sure nothing gets missed

**Pull request merged**